corehq/apps/linked_domain/view_helpers.py

```diff
@@ -315,8 +353,23 @@ def pop_keyword_for_action(action, keywords):
     return keyword
 
 
-def build_pullable_view_models_from_data_models(domain, upstream_link, apps, fixtures, reports, keywords,
-                                                timezone, is_superuser=False):
+def pop_ucr_expression_for_action(action, ucr_expressions):
+    ucr_expression_id = action.wrapped_detail.ucr_expression_id
+    try:
+        ucr_expression = ucr_expressions[ucr_expression_id]
+        del ucr_expressions[ucr_expression_id]
+    except KeyError:
```

Comment:
I think this can be simplified with (which also avoids a nested try-catch)
```
ucr_expression  = ucr_expressions.pop(ucr_expression_id, None)
```

Reply:
You are right! I just copied the way this was done for the other items in this file and didn't think too hard about it. I will update.

Reply:
https://github.com/dimagi/commcare-hq/pull/31597/commits/55dba1503f7d3ddc02634044a248a0391286986e